microsoft/testsuites/network/networksettings.py

```diff
@@ -563,75 +567,117 @@ def validate_device_statistics(self, environment: Environment, log: Logger) -> N
+                    if pkts == 0
                 ]
 
-                per_rx_queue_packets = [
-                    v
-                    for (k, v) in device_stats.counters.items()
-                    if self._rx_queue_stats_regex.search(k)
+                starving_rx_queues = [
+                    queue
+                    for queue, pkts in enumerate(per_rx_queue_packets)
+                    if pkts == 0
                 ]
 
-            avg_tx_queue_packets = sum(per_tx_queue_packets) / len(per_tx_queue_packets)
-            per_tx_queue_pkt_percent: List[float] = []
-            for packet_count in per_tx_queue_packets:
-                per_tx_queue_pkt_percent.append(packet_count / avg_tx_queue_packets)
+                starved_queues_per_device[device_stats.interface] = (
+                    starving_tx_queues,
+                    starving_rx_queues,
+                )
 
-            assert_that(
-                (max(per_tx_queue_pkt_percent) - min(per_tx_queue_pkt_percent)),
-                f"{an_enabled}Statistics show traffic is not evenly distributed"
-                " among tx queues",
-            ).is_less_than_or_equal_to(0.5)
+            for k in starved_queues_per_device:
+                if k in prev_starved_queues_per_device:
+                    txq_list, rxq_list = starved_queues_per_device[k]
+                    prev_txq_list, prev_rxq_list = prev_starved_queues_per_device[k]
+                    final_starved_txq = [
+                        txq for txq in txq_list if txq in prev_txq_list
+                    ]
+                    final_starved_rxq = [
+                        rxq for rxq in rxq_list if rxq in prev_rxq_list
+                    ]
+
+            if (
+                not prev_starved_queues_per_device
+                and (not final_starved_txq)
+                and (not final_starved_rxq)
+            ):
+                raise PassedException(
+                    f"No starved queues found by running test {counter} times."
+                )
 
-            avg_rx_queue_packets = sum(per_rx_queue_packets) / len(per_rx_queue_packets)
-            per_rx_queue_pkt_percent: List[float] = []
-            for packet_count in per_rx_queue_packets:
-                per_rx_queue_pkt_percent.append(packet_count / avg_rx_queue_packets)
+            prev_starved_queues_per_device.clear()
+            prev_starved_queues_per_device = copy.deepcopy(starved_queues_per_device)
+            starved_queues_per_device.clear()
 
-            assert_that(
-                (max(per_rx_queue_pkt_percent) - min(per_rx_queue_pkt_percent)),
-                f"{an_enabled}Statistics show traffic is not evenly distributed"
-                " among rx queues",
-            ).is_less_than_or_equal_to(0.5)
+            counter += 1
+
+        assert_that(
+            final_starved_txq,
+            f"The tx stats for queue/queues {final_starved_txq} is/are 0.",
```

Comment:
Imaging if the test case failed by this assertion, and the user won't look into the code. What can she/he know about the failure, and how should it be fixed?

Reply:
This would need deeper investigation but for now we would know that any particular queue is being starved. Ok I can add message like Need further investigation in driver code.

Reply:
Not seeing changes to make it clearer.